mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -262,10 +267,17 @@ pub(crate) async fn block_header_utxo_loop<T: UtxoCommonOps>(
 
         let storage = client.block_headers_storage();
         let from_block_height = match storage.get_last_block_height().await {
-            Ok(h) => h,
-            Err(e) => {
-                error!("Error {e:?} on getting the height of the last stored {ticker} header in DB!",);
-                sync_status_loop_handle.notify_on_temp_error(e.to_string());
+            Ok(Some(height)) => height,
+            Ok(None) => {
+                if let Err(e) = validate_and_store_starting_header(client, ticker, storage, &spv_conf).await {
+                    sync_status_loop_handle.notify_on_permanent_error(e);
+                    break;
```

Comment:
> So in this case we should retry at the next iteration

No need for retry in my opinion, since this will return an error quickly on coin activation because it's a permanent error.

Reply:
I see, thanks!